Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -509,6 +509,83 @@ $ adb install UnCrackable-Level1.objection.apk
 $ objection explore
 ```
 
+#### Basic Information Gathering
+
+As mentioned previously, Android runs on top of a modified Linux kernel. Android retains the [proc filesystem](https://www.kernel.org/doc/Documentation/filesystems/proc.txt "procfs") from Linux, and this filesystem is mounted at /proc.  The proc filesystem (procfs) provides a directory-based view of process running on the system, providing information about the processes, threads, and other system-wide diagnostics. Procfs is arguably one of the most important filesystems on Android. Many Android tools depend on procfs as the source of information.
+
+Many command line tools are not shipped with the Android firmware to reduce the size, but can be easily installed on a rooted device using BusyBox. We can also create our own custom scripts using commands like `cut`, `grep`, `sort` etc, to parse the proc filesystem information.
+
+In this section we will be using information from the proc filesystem directly or indirectly to gather information about a running process.
+
+##### Open Files
+
+`lsof` can provide a list of all open files, including a stream, a network file or a regular file. When invoking the `lsof` command without any option it will list all open files belonging to all active processes on the system, while when invoking with the flag `-p <pid>`, it returns the list of open files for the specified process. The [man page](http://man7.org/linux/man-pages/man8/lsof.8.html "Man Page of lsof") shows various other options in detail.
+
+```
+angler:/ # lsof -p 6233
+COMMAND     PID       USER   FD      TYPE             DEVICE  SIZE/OFF       NODE NAME
+.foobar.c  6233     u0_a97  cwd       DIR                0,1         0          1 /
+.foobar.c  6233     u0_a97  rtd       DIR                0,1         0          1 /
+.foobar.c  6233     u0_a97  txt       REG             259,11     23968        399 /system/bin/app_process64
+.foobar.c  6233     u0_a97  mem   unknown                                         /dev/ashmem/dalvik-main space (region space) (deleted)
+.foobar.c  6233     u0_a97  mem       REG              253,0   2797568    1146914 /data/dalvik-cache/arm64/system@framework@boot.art
+.foobar.c  6233     u0_a97  mem       REG              253,0   1081344    1146915 /data/dalvik-cache/arm64/system@framework@boot-core-libart.art
+.foobar.c  6233     u0_a97  mem       REG              253,0    307200    1146916 /data/dalvik-cache/arm64/system@framework@boot-conscrypt.art
+.foobar.c  6233     u0_a97  mem       REG              253,0    225280    1146917 /data/dalvik-cache/arm64/system@framework@boot-okhttp.art
+....
+```
+
+##### Open Connections
+
+The directory `/proc/<PID>/net` provides networking information, but the information in this file is not process specific. Instead, it is the same as information as `/proc/net`. There are multiple files present in this directory, of which `tcp`, `tcp6` and `udp` can be considered important from our perspective.
```

Comment:
Please take a look at this suggestion, I tried to make it a bit more concise. Apart from that I have one question: Is it really the output of cat /proc/7254/net/tcp the same as the output of cat /proc/net/tcp?
```suggestion
You can find system-wide networking information in `/proc/net` or just by inspecting the `/proc/<pid>/net` directories (for some reason not process specific). There are multiple files present in these directories, of which `tcp`, `tcp6` and `udp` might be considered relevant from the tester's perspective.
```

Reply:
As per the documentation of /proc, that is the case. It is counter-intuitive, but that why I get from other comments on various forums as well. 
http://man7.org/linux/man-pages/man5/proc.5.html